intelmq/bots/collectors/kafka/collector.py

```diff
@@ -0,0 +1,45 @@
+"""
+Kafka Collector Bot
+
+Collects information from the Apache Kafka distributed stream processing system.
+
+Args:
+    topic (str): topic to collect information from
+    bootstrap_servers (str): the ‘host[:port]’ string of the Apache Kafka system. Defaults to `localhost:9092`
+"""
+
+from intelmq.lib.bot import CollectorBot
+from intelmq.lib.exceptions import MissingDependencyError
+
+try:
+    import kafka
+except ImportError:
+    kafka = None
+
+class KafkaCollectorBot(CollectorBot):
+
+    def init(self):
+        if kafka is None:
+            raise MissingDependencyError("kafka")
+
+        self.topic = []
+        if getattr(self.parameters, 'topic', '') != '':
+            self.topic = self.parameters.topic
+
+        self.bootstrap_servers = 'localhost:9092'
+        if getattr(self.parameters, 'bootstrap_servers', '') != '':
+            self.bootstrap_servers = self.parameters.bootstrap_servers
+
+        self.logger.debug(f"topic set to {self.topic}, bootstrap_servers set to {self.bootstrap_servers}")
+
+    def process(self):
+        " Fetch messages from the Kafka server and pass them on one by one "
+        consumer = kafka.KafkaConsumer(self.topic, bootstrap_servers=self.bootstrap_servers, auto_offset_reset='earliest', consumer_timeout_ms=100, group_id=self._Bot__bot_id)
+        for msg in consumer:
+            consumer.commit()
+            report = self.new_report()
+            self.logger.debug(f"Received msg with offset {msg.offset} from kafka topic {msg.topic}: {msg.value.decode()}")
```

Comment:
that requires python 3.6, but intelmq supports 3.5 as well. See also the test fail: https://travis-ci.org/github/certtools/intelmq/builds/742395055

Instead of requiring a different python version for this bot and excluding it from the tests, I propose not using format strings as long as 3.5 is supported.

Reply:
fixed